webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/isNodeUsed.scala.txt

```diff
@@ -0,0 +1,52 @@
+@*
+ * Copyright © 2015-2019 the contributors (see Contributors.md).
+ *
+ * This file is part of Knora.
+ *
+ * Knora is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Knora is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with Knora.  If not, see <http://www.gnu.org/licenses/>.
+ *@
+
+@import org.knora.webapi.IRI
+
+@*
+ * Checks whether a node is used (i.e. is the object of any statements).
+ *
+ * @param triplestore the name of the triplestore being used.
+ * @param nodeIri the IRI of the node.
+ *@
+@(triplestore: String,
+  nodeIri: IRI)
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+PREFIX salsah-gui: <http://www.knora.org/ontology/salsah-gui#>
+
+SELECT DISTINCT ?isUsed
+@* Ensure that inference is not used in this query. *@
+@if(triplestore.startsWith("graphdb")) {
+    FROM <http://www.ontotext.com/explicit>
+}
+WHERE {
+  	BIND(IRI("@nodeIri") AS ?node)
+    BIND(true AS ?isUsed)
+	{
+    	?s ?p "hlist=<@nodeIri>" .
+  	} UNION {
+  		?s ?p ?node .
+  	}
+  FILTER((?p = knora-base:valueHasListNode) || (?p = salsah-gui:guiAttribute))
```

Comment:
Why not put these properties in the statements in the `UNION`, instead of using a `FILTER`?

Reply:
Good point! I kept modifying this template so often, I forgot to optimize at the end when it worked.